roles/neutron-data/tasks/main.yml

```diff
@@ -49,6 +49,18 @@
     - ursula_os == 'rhel'
     - neutron.plugin == 'ml2'
 
+- name: mask unwanted neutron services on compute (rhel)
+  command: systemctl mask {{ item.name }}
+  with_items:
+    - "{{ neutron.services.neutron_server }}"
+    - "{{ neutron.services.neutron_dhcp_agent }}"
+    - "{{ neutron.services.neutron_l3_agent }}"
+    - "{{ neutron.services.neutron_metadata_agent }}"
+  when:
+    - ursula_os == 'rhel'
+    - inventory_hostname in groups['compute']
```

Comment:
right, current when condition will also mask neutron services on controllers (which are part of network group), right?
So we should be changing when as u mentioned in ur comment.